internal/controllers/synthesis/lifecycle_test.go

```diff
@@ -819,11 +819,19 @@ func TestShouldSwapStates(t *testing.T) {
 			},
 		},
 	}
+
 	for _, tc := range tests {
 		t.Run(tc.Name, func(t *testing.T) {
 			syn := &apiv1.Synthesizer{}
 			syn.Spec.Refs = []apiv1.Ref{{Key: "foo"}}
-			assert.Equal(t, tc.Expectation, shouldSwapStates(syn, &tc.Composition))
+			reason, shouldSwap := shouldSwapStates(syn, &tc.Composition)
```

Comment:
Should we add the expected reason to the test cases?

Reply:
change to verify the reason